server/room-battle.ts

```diff
@@ -1338,6 +1353,367 @@ export class RoomBattle extends RoomGames.RoomGame<RoomBattlePlayer> {
+			}
+		}
+		if (this.games.length >= this.bestOf) return this.onEnd(''); // tie
+		// no one has won, skip onwards
+		setImmediate(() => this.promptNextGame(room));
+	}
+	promptNextGame(room: Room) {
+		if (!room.battle || this.winner) return; // ???
+		const cmd = `/msgroom ${this.room.roomid},/confirmready`;
+		for (const userid in room.battle.playerTable) {
+			const player = room.battle.playerTable[userid];
+			player.id = userid as ID; // re-link users so that we can use timer properly
+			const name = Utils.escapeHTML(this.name(userid));
+			const button = `|c|&|/uhtml prompt-${userid},<button class="button notifying" name="send" value="${cmd}">I'm ready!</button>`;
+			const prompt = `|c|&|/log Are you ready for game ${this.games.length + 1}, ${name}?`;
+			player.sendRoom(prompt);
+			player.sendRoom(button);
+			// send it to the main room as well, in case they x out of the old one
+			this.playerTable[userid].sendRoom(prompt);
+			this.playerTable[userid].sendRoom(button);
+		}
+		this.waitingBattle = room.battle;
+		this.ready = {p1: false, p2: false};
+		this.nextBattleTimerStart = Date.now();
+		this.nextBattleTimer = setInterval(() => this.pokeNextBattleTimer(), 10_000);
+	}
+	pokeNextBattleTimer() {
+		if (!this.nextBattleTimerStart || !this.nextBattleTimer) return; // ??
+		if ((Date.now() - this.nextBattleTimerStart) >= (60_000)) {
+			return this.nextGame();
+		}
+		for (const k of ['p1', 'p2'] as const) {
+			if (!this.ready![k]) {
+				const diff = (this.nextBattleTimerStart + 60000) - Date.now();
+				this.waitingBattle?.room.add(
+					`|inactive|${this.name(this[k])} has ${Chat.toDurationString(diff + 1000)}` +
+					` to confirm battle start!`
+				);
+			}
+		}
+		this.waitingBattle?.room.update();
+		this.room.update();
+	}
+	confirmReady(user: ID) {
+		if (![this.p1, this.p2].includes(user)) {
+			throw new Chat.ErrorMessage("You aren't a player in this best-of set.");
+		}
+		const battle = this.waitingBattle;
+		if (!this.ready || !battle) {
+			throw new Chat.ErrorMessage("The battle is not currently waiting for ready confirmation.");
+		}
+
+		this.ready[user === this.p1 ? 'p1' : 'p2'] = true;
+		const readyMsg = Utils.html`|c|&|/uhtml prompt-${user},${this.name(user)} is ready for game ${this.games.length + 1}!`;
+		battle.room.add(readyMsg).update();
+		if (Object.values(this.ready).filter(Boolean).length === 2) {
+			this.nextGame();
+		}
+	}
+	getLatestBattle() {
+		// Strictly, this should never be null unless battle creation lags big time. but let's be typesafe anyway.
+		return this.games[this.games.length - 1].battle.battle;
+	}
+	private name(str: string) {
+		return Users.get(str)?.name || str;
+	}
+	async onEnd(winner: ID) {
+		this.cleanup();
+		this.room.add(`|allowleave|`).update();
+		if (winner) {
+			this.winner = winner;
+			this.room.add(Utils.html`|html|<h2>${this.name(winner)} wins!</h2>`);
+		} else {
+			this.winner = '';
+			this.room.add(`The battle was tied.`);
+		}
```

Comment:
Why not use `|win` and `|tie` here instead of sending a custom message? Would make sure the forfeit popup doesn't appear after the set is over on the client.

Reply:
Oh yeah, forgot those existed. Ty!